tests/foreman/cli/test_docker.py

```diff
@@ -37,50 +37,101 @@
 
+logger = logging.getLogger('robottelo')
 
-def _make_docker_repo(product_id, name=None, upstream_name=None, url=None):
+
+def _repo(product_id, name=None, upstream_name=None, url=None):
     """Creates a Docker-based repository.
 
     :param product_id: ID of the ``Product``.
     :param str name: Name for the repository. If ``None`` then a random
         value will be generated.
     :param str upstream_name: A valid name of an existing upstream repository.
-        If ``None`` then defaults to ``busybox``.
+        If ``None`` then defaults to DOCKER_UPSTREAM_NAME constant.
     :param str url: URL of repository. If ``None`` then defaults to
         DOCKER_REGISTRY_HUB constant.
     :return: A ``Repository`` object.
     """
     return make_repository(
         {
-            'content-type': REPO_CONTENT_TYPE,
-            'docker-upstream-name': upstream_name or REPO_UPSTREAM_NAME,
-            'name': name or choice(generate_strings_list(15, ['numeric', 'html'])),
+            'content-type': REPO_TYPE['docker'],
+            'docker-upstream-name': upstream_name or DOCKER_UPSTREAM_NAME,
+            'name': name or gen_string('alpha', 5),
             'product-id': product_id,
             'url': url or DOCKER_REGISTRY_HUB,
         }
     )
 
 
-class DockerManifestTestCase(CLITestCase):
+def _content_view(repo_id, org_id):
+    """Create a content view and link it to the given repository."""
+    content_view = make_content_view({'composite': False, 'organization-id': org_id})
+    ContentView.add_repository({'id': content_view['id'], 'repository-id': repo_id})
+    return ContentView.info({'id': content_view['id']})
+
+
+@pytest.fixture
+def docker_host(rhel7_contenthost):
+    """Instantiate and set up a docker host VM. Destroy VM when done."""
+    logger.info('Installing katello-ca on the external docker host')
+    rhel7_contenthost.install_katello_ca()
+
+    repos = {'server': SERVER_REPO, 'optional': OPTIONAL_REPO, 'extras': EXTRAS_REPO}
+    rhel7_contenthost.create_custom_repos(**repos)
+    rhel7_contenthost.execute('yum -y install docker')
+    rhel7_contenthost.execute('systemctl start docker')
+    return rhel7_contenthost
+
+
+@pytest.fixture
+def repo(module_product):
+    return _repo(module_product.id)
+
+
+@pytest.fixture
+def content_view(module_org, repo):
+    return _content_view(repo['id'], module_org.id)
+
+
+@pytest.fixture
+def content_view_publish(content_view):
+    ContentView.publish({'id': content_view['id']})
+    content_view = ContentView.info({'id': content_view['id']})
+    return ContentView.version_info({'id': content_view['versions'][0]['id']})
+
+
+@pytest.fixture
+def content_view_promote(content_view_publish, module_lce):
+    ContentView.version_promote(
+        {
+            'id': content_view_publish['id'],
+            'to-lifecycle-environment-id': module_lce.id,
+        }
+    )
+    return ContentView.version_info({'id': content_view_publish['id']})
```

Comment:
For test setup purpose you can use nailgun existing fixtures as they are faster than CLI.